src/hotspot/share/cds/archiveBuilder.cpp

```diff
@@ -589,6 +591,21 @@ char* ArchiveBuilder::ro_strdup(const char* s) {
   return archived_str;
 }
 
+int ArchiveBuilder::compare_src_objs(SourceObjInfo** a, SourceObjInfo** b) {
```

Comment:
I think some explanation would be useful:

```
  // The objects that have embedded pointers will be bubbled up
  // towards the top of the list. This ensures we have a maximum
  // number of leading zero bits in the relocation bitmap.
  if ((*a)->has_embedded_pointer() && !(*b)->has_embedded_pointer()) {
    return 1;
  } else if (!(*a)->has_embedded_pointer() && (*b)->has_embedded_pointer()) {
    return -1;
  } else {
    // This is necessary to keep the sorting order stable. Otherwise the
    // archive's contents may not be deterministic.
    return (*a)->index() - (*b)->index();
  }


```